test/gpu/native/basics/outOfOnArr.chpl

```diff
@@ -0,0 +1,34 @@
+use GpuDiagnostics;
+
+config const n = 10;
+
+class ArrWrapperClass {
+  const Dom: domain(?);
+  forwarding var Arr: [Dom] int;
+}
+
+type ArrWrapper = owned ArrWrapperClass?;
+
+startGpuDiagnostics();
+
+var HostArr = new ArrWrapper({1..n});
+var DevArr: HostArr.type; // make sure to not use a generic type
+
+on here.gpus[0] {
+  DevArr = new ArrWrapper({1..n}); // allocates the array on the GPU
+
+  foreach a in DevArr! do
+    a += 1;
+}
+
+on here.gpus[0] {
+  foreach a in DevArr! do
```

Comment:
Should this loop utilize an "assert on GPU" to lock in that we're not running on the GPU?  That's pretty cool that the use of the class doesn't break GPU-ization.

Reply:
Hmm.. I am on the fence. `assertGpuDiags` captures that for our testing purposes. `assertOnGpu` is really useful, but not sure if I want it to proliferate. I am probably overthinking, though. I can add it here.

Reply:
Ah, OK, I'm good with that.  I was looking way too localized to notice the diags call—thanks!